fuel/datasets/hdf5.py

```diff
@@ -275,12 +270,11 @@ def subsets(self):
                 if num_examples != subset.stop - subset.start:
                     raise ValueError("sources have different lengths")
             self._subsets = subsets
-            self._out_of_memory_close(handle)
         return self._subsets
 
     def load(self):
         if self.load_in_memory:
-            handle = self._out_of_memory_open()
```

Comment:
In this specific case you should open and close: we're reading the file from disk and into memory. This is probably why your tests are failing.